db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -352,6 +380,26 @@ public <T> void reschedule(TaskInstanceId taskInstanceId, Instant newExecutionTi
       }
     }
 
+    @Override
+    public <T> void rescheduleOrCreate(TaskInstance<T> taskInstance, Instant executionTime) {
+      Optional<ScheduledExecution<Object>> executionObject =
+          this.getScheduledExecutions().stream()
+              .filter(e -> e.getTaskInstance().getId().equals(taskInstance.getId()))
+              .findFirst();
```

Comment:
We should avoid fetch all here, instead use
```
Optional<ScheduledExecution<Object>> getScheduledExecution(TaskInstanceId taskInstanceId);
```

Reply:
You are right about that. I have adopted your suggestion.